src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -305,27 +304,32 @@ internal LocationVersionOne CreateLocation(Location v2Location)
                     ResultFile = CreatePhysicalLocation(v2Location.PhysicalLocation)
                 };
 
-                if (!string.IsNullOrWhiteSpace(v2Location.FullyQualifiedLogicalName) &&
-                    _currentV2Run.LogicalLocations?.ContainsKey(v2Location.FullyQualifiedLogicalName) == true)
+                if (!string.IsNullOrWhiteSpace(v2Location.FullyQualifiedLogicalName))
                 {
-                    location.DecoratedName = _currentV2Run.LogicalLocations[v2Location.FullyQualifiedLogicalName].DecoratedName;
+                    location.DecoratedName = _currentV2Run.LogicalLocations[v2Location.LogicalLocationIndex].DecoratedName;
                 }
             }
 
             return location;
         }
 
         internal LogicalLocationVersionOne CreateLogicalLocation(LogicalLocation v2LogicalLocation)
-        {
+        {            
             LogicalLocationVersionOne logicalLocation = null;
+            string parentKey = null;
+
+            if (_currentV2Run.LogicalLocations != null && v2LogicalLocation.ParentIndex > -1)
+            {
+                parentKey = _currentV2Run.LogicalLocations[v2LogicalLocation.ParentIndex].FullyQualifiedName;
```

Comment:
>FullyQualifiedName [](start = 90, length = 18)

BUG: Again, since FQLN is optional if it's the same as name, you need FQLN ?? Name.

Reply:
[https://github.com/Microsoft/sarif-sdk/issues/1181](https://github.com/Microsoft/sarif-sdk/issues/1181) marked urgent. suggest just creating a simple helper for constructing the key name

---
In reply to: [243867117](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r243867117) [](ancestors = 243867117)